node/src/components/deploy_acceptor.rs

```diff
@@ -103,21 +115,32 @@ impl DeployAcceptor {
         deploy: Box<Deploy>,
         source: Source<NodeId>,
         deploy_config: DeployAcceptorConfig,
+        maybe_responder: Option<Responder<Result<(), Error>>>,
     ) -> Effects<Event> {
         let mut cloned_deploy = deploy.clone();
+        let mut effects = Effects::new();
         if is_valid(&mut cloned_deploy, deploy_config) {
-            effect_builder
-                .put_deploy_to_storage(cloned_deploy)
-                .event(move |is_new| Event::PutToStorageResult {
+            if let Some(responder) = maybe_responder {
+                effects.extend(responder.respond(Ok(())).ignore());
+            }
+            effects.extend(effect_builder.put_deploy_to_storage(cloned_deploy).event(
+                move |is_new| Event::PutToStorageResult {
                     deploy,
                     source,
                     is_new,
-                })
+                },
+            ));
         } else {
-            effect_builder
-                .announce_invalid_deploy(deploy, source)
-                .ignore()
+            if let Some(responder) = maybe_responder {
```

Comment:
Add a comment to explain why you are doing this please, for the benefit of the next person to work in this file...which might even be future you!

Reply:
Added comments